crates/router/src/core/utils.rs

```diff
@@ -599,7 +599,7 @@ pub fn is_merchant_enabled_for_payment_id_as_connector_request_id(
     let config_map = &conf
         .connector_request_reference_id_config
         .merchant_ids_send_payment_id_as_connector_request_id;
-    config_map.contains(merchant_id)
+    config_map.iter().any(|id| id == merchant_id)
```

Comment:
what's the difference?

Reply:
contains expects &String, so had to convert &str to &String. To avoid that, changed it to iter any logic